### DISCUSSION TITLE
JS Error on Apache2 SecurityError: Permission denied to access property "document" on cross-origin object

### DISCUSSION THREAD

**Discussion by KenDrummond** [Novik/ruTorrent#2481] [Q&A]
I have recently upgraded to ubuntu 20.04 with apache2 2.4.41 and I am receiving a JS Error when attempting to load a magnet link through ruTorrent. I get the same error with ruTorrent 3.9 and 4.0.4.  

The error is "[17.04.2023 21:44:13] JS error: [https://www.XXXXXXXXXXXXX.com/rutorrent/js/content.js?v=404 : 81] SecurityError: Permission denied to access property "document" on cross-origin object" 

I think this is an issue with apache2, but I have tried a few apache2 config options but still get this error.  Any ideas?

**Comment by stickz** [comment]
Have you tried cleaning your web browser cache? 

Did you perform a fresh install of ruTorrent 4.0.4 with the new version of config.php? Using the default configuration?
https://github.com/Novik/ruTorrent/blob/master/conf/config.php

**Reply by KenDrummond** [reply]
Thanks for the response, yes this was a clean install of 4.0.4 from git i'm pretty sure this is an apache2 issue related to XSS.  I get the same (similar) error with 3.9 and 4.0.4.

Do you not get this error when adding magnet links under apache 2.4.41?

**Reply by KenDrummond** [reply]
I have tried with both firefox and chrome browsers.

**Comment by stickz** [comment]
Are you able to try an alternate web server such as nginx? Also, I would highly recommend using 4.0.4, version 3 is no longer supported.

If you're really stuck, swizzin has an install script that I've submitted improvements to.
https://github.com/swizzin/swizzin

There's also a less optimized docker option, if changes at the operating system layer are problematic. But this has disadvantages.
https://github.com/crazy-max/docker-rtorrent-rutorrent

**Reply by KenDrummond** [reply]
Thanks, I have installed 4.0.4 and get the same error, I'd really like to get this working on apache2 I have a few small pages that run under apache2. I did have to refactor my self developed Asterisk Report page to stop using frames when I brought it across to ubuntu 20.04 and apache2 2.4.41.

**Reply by stickz** [reply]
There isn't much we can do to help you install `ruTorrent` on your own. This is why I encourage using an install script, if you're unable to get things working properly. I provided you with two suggestions. I've tested and contributed to both scripts to help make them a more viable option for end users. Depending on your situation, one might be more feasible than the other.

**Reply by KenDrummond** [reply]
I have run an older version or rutorrent on an older version of apache2 for many years and the only thing that seems not to be working on an updated version of apache2 is adding magnet links though rutorrent. 

Can you confirm that adding magnet links through rutorrent works for you with a recent version of apache2?

**Reply by KenDrummond** [reply]
As I said the issue seems to be with apache2 and how rutorrent is interacting with it in an apparently non cross site safe way (at least according to apache).

**Reply by stickz** [reply]
Please file a very detailed issue report about your configuration, if you believe there is a bug with magnet links on apache2.
